processing/src/main/java/io/druid/segment/data/ColumnarDoubles.java

```diff
@@ -54,6 +55,44 @@ public double getDouble(int offset)
         return ColumnarDoubles.this.get(offset);
       }
 
+      @Override
+      public boolean isNull()
+      {
+        return false;
+      }
+
+      @Override
+      public void inspectRuntimeShape(RuntimeShapeInspector inspector)
+      {
+        inspector.visit("columnar", ColumnarDoubles.this);
+        inspector.visit("offset", offset);
+      }
+    }
+    return new HistoricalDoubleColumnSelector();
+  }
+
+  default ColumnValueSelector<Double> makeColumnValueSelector(ReadableOffset offset, ImmutableBitmap nullValueBitmap)
```

Comment:
I think it's safer when there is just one method, and it checks if the bitmap is empty internally, and if it is, it returns optimized implementation. Otherwise it's easier to make a mistake in some other places, and also it creates two overloaded methods which differ by the presence of the last parameter, so you may want to pass it, but just forget, but the code will still compile and run. Same comments for long and float.

Reply:
done.